### DISCUSSION TITLE
This usually means that your host's kernel does not support WireGuard!

### DISCUSSION THREAD

**Discussion by 897896639** [wg-easy/wg-easy#1514] [Q&A]
operating system：openEuler 24.03
kernel:6.6.0-28.0.0.34.oe2403.x86_64

![image](https://github.com/user-attachments/assets/931560cb-fe5e-4693-b4e8-ada10f944ad4)

`2024-11-13T01:42:56.238Z Server Listening on http://0.0.0.0:51821
2024-11-13T01:42:56.253Z WireGuard Loading configuration...
2024-11-13T01:42:56.258Z WireGuard Configuration loaded.
2024-11-13T01:42:56.258Z WireGuard Config saving...
2024-11-13T01:42:56.261Z WireGuard Config saved.
$ wg-quick down wg0
$ wg-quick up wg0
Error: WireGuard exited with the error: Cannot find device "wg0"
This usually means that your host's kernel does not support WireGuard!
    at /app/lib/WireGuard.js:74:17
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async WireGuard.getConfig (/app/lib/WireGuard.js:72:7)`

**Comment by 897896639** [comment]
我使用openEuler 2203sp4也是报同样的错误，openEuler 2203sp4的内核为5.10

**Reply by kaaax0815** [reply]
Please write in english.
I am not familiar with openEuler but it seems like Wireguard packages are available.

try running
```shell
modprobe wireguard
```
before running the container.

**Reply by dhoodx64** [reply]
I have this exact same issue with Ubuntu 24. This did not work and I am using a supported kernel.  Wondering if you or anyone else has any idea how to resolve this. I am using this on k3s but I should at least be able to get the pod started. I can provide more details if needed.

**Reply by dhoodx64** [reply]
> I have this exact same issue with Ubuntu 24. This did not work and I am using a supported kernel. Wondering if you or anyone else has any idea how to resolve this. I am using this on k3s but I should at least be able to get the pod started. I can provide more details if needed.

Nevermind past me is blind, I found the below on the wiki on running wgeasy with Kubernetes. I assume @897896639 is having the same issue, needs to allow the container to interact with the host OS kernel. When I get some time I can see if I can dig the option up for docker

# Give pod permissions to modify iptables and load the wireguard kernel module
          securityContext:
            capabilities:
              add:
                - NET_ADMIN
                - SYS_MODULE
